flyteadmin/pkg/server/service.go

```diff
@@ -422,11 +426,34 @@ func serveGatewayInsecure(ctx context.Context, pluginRegistry *plugins.Registry,
 		ReadHeaderTimeout: time.Duration(cfg.ReadHeaderTimeoutSeconds) * time.Second,
 	}
 
-	err = server.ListenAndServe()
-	if err != nil {
-		return errors.Wrapf(err, "failed to Start HTTP Server")
+	go func() {
+		err = server.ListenAndServe()
+		if err != nil && err != http.ErrServerClosed {
+			logger.Fatalf(ctx, "Failed to start HTTP Server: %v", err)
+		}
+	}()
+
+	// Gracefully shutdown the servers
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
+	<-sigCh
+	time.Sleep(1 * time.Second)
+
+	// force to shut down servers after 10 seconds
+	timer := time.AfterFunc(10 * time.Second, func() {
```

Comment:
May want to make this timeout configurable

Reply:
Like configure the timeout in flyteadmin_config.yaml?

```
server:
  forceShutdownTimeoutSec: 10

Reply:
Yeah. Something along those lines.

Reply:
updated